keps/sig-cluster-lifecycle/generic/1755-communicating-a-local-registry/README.md

```diff
@@ -0,0 +1,618 @@
+For example, consider both security and how this will impact the larger
+kubernetes ecosystem.
+
+How will security be reviewed and by whom?
+
+How will UX be reviewed and by whom?
+
+Consider including folks that also work outside the SIG or subproject.
+-->
+
+None
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable.  This may include API specs (though not always
+required) or even code snippets.  If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+### The `local-image-loaders` ConfigMap
+
+Tools that configure a local registry should apply a ConfigMap to the cluster.
+
+Documentation that educates people on how to set up a registry manually should
+include instructions on what the ConfigMap should be.
+
+The name of the ConfigMap must be `local-image-loaders`.
+
+The namespace must be `kube-public`.
+
+Each field of the ConfigMap documents a way to load images into the cluster.
+This proposal only specifies one field.
+
+When a local cluster adds support for new ways to import an image that they want
+to advertise (e.g., `kind load`), they should propose new fields to the
+top-level ConfigMap.
+
+Example:
+
+```
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: local-image-loaders
+  namespace: kube-public
+```
+
+### The `localRegistryStatus.v1` field
+
+The `localRegistryStatus.v1` field of the ConfigMap specifies the registry
+details.  When specified, the field must contain YAML.
+
+The contents of a `localRgistryStatus.vX` specification are frozen and will not
+change.
+
+When adding, removing or renaming fields, future proposals should increment the
+MAJOR version of an API hosted in the ConfigMap - e.g. `localRegistryStatus.vX`
+where X is incremented.
+
+Writers of this ConfigMap (i.e., cluster configuration tooling) should write as
+many top-level fields as the cluster supports.  The most recent version is the
+source of truth.
+
+Readers of this ConfigMap should start with the most recent version they support
+and work backwards. Readers are responsible for doing any defaulting of fields
+themselves without the assistance of any common API machinery.
+
+Example data:
+
+```
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: local-image-loaders
+  namespace: kube-public
+data:
+  localRegistryStatus.v1: |
+    host: "localhost:5000"
+    hostFromCluster: "registry:5000"
+    help: "https://kind.sigs.k8s.io/docs/user/local-registry/"
+```
+
+The following code is the reference implementation that readers may use to parse the YAML:
+
+```go
+
+// LocalRegistryStatusV1 describes a local registry that developer tools can
+// connect to. A local registry allows clients to load images into the 
+// local cluster by pushing to this registry.
+type LocalRegistryStatusV1 struct {
+	// Host documents the host (hostname and port) of the registry, as seen from
```

Comment:
forgot to ask, i'm assuming this can be "hostname/ip and port" too?

Reply:
My understanding (from reading the OCI scope charter) is that the format of this is controversial and deliberately not well-specified. I suspect that putting an IPv4
address here will work on all container runtime implementations. I'm less sure whether an IPv6 address will work.

Reply:
ok, searching around for similar documentation "hostname and port" seems commonly used, so lets leave it like that.